### ISSUE TITLE
Unable to set name and manufacturer

### ISSUE THREAD

**Issue by StevePER** [volks73/cargo-wix#137]
Firstly thanks for developing cargo-wix, it seems like it will significantly simplify packaging my software for Windows.

I would like to change the name and manufacturer that are shown in the installer. Obviously I can edit main.wxs but [the documentation](https://volks73.github.io/cargo-wix/cargo_wix/index.html) says I can also specify these parameters via the command line or by adding a [package.metadata.wix] section to Cargo.toml. I've tried both of these methods unsuccessfully.

`cargo wix --name MyPreferredName` seems to do nothing. The --manufacturer argument produces an error.

In Cargo.toml, I've tried:
```
[package.metadata.wix]
name = "MyPreferredName"
manufacturer = "My Company Name"
```

Neither of these options seems to do anything.

**Comment by volks73** [comment]
Thank you for your interest in `cargo-wix` and the kind words. I am sorry you are experiencing some issues with it.

The `-m, --manufacturer` option is only available with the `cargo wix init` or `cargo wix print` commands:

>Available for the _init_ (cargo wix init) and _print_ (cargo wix print) subcommands.

In other words, it is only useful when creating the Main.wxs file and not when creating the installer (MSI) with the `cargo wix` (default) command. The idea being that the manufacturer should not be changing between installer creations. It should always be the same entity/person/organization that is manufacturing the product/application packaged into the installer.

Since the `-m,--manufacturer` option is only available for the _init_ and _print_ subcommands, the `manufacturer` field in the `[package.metadata.wix]` section does not work. The `[package.metadata.wix]` section only applies to the _create_ (default) subcommand.

However, the `-n,--name` option is available for the `cargo wix` (default/_create_) command:

>Available for the default _create_ (cargo wix) subcommand.

So, the `name` field in the `[package.metadata.wix]` section should work. Note, this only names the name of the installer, not the executable that is installed. For example, the `-n,--name` option or the `name` field will change `FOO-1.2.3-x86_64.msi` to `BAR-1.2.3-x86_64.msi`, while the executable (cargo built target with the binary name defined in the Cargo.toml file) will be unchanged in its name.

Can you confirm that there is no effect wit the `-n,--name` option or field? If there is no change in the created installer's file stem, then this is a bug.

Regardless, I did notice there is no integration test with the `-n,--name` option or metadata field. I will investigate further on any tests related to this feature as it having no effect should have been caught by a test.

**Comment by StevePER** [comment]
Both `-m` and `-n` options work correctly as you have specified above. On re-reading the documentation it's quite clear that the `-m` option needs to be specified with the `init` subcommand.

Although not specified in the documentation, I had expected the `name` option to override the Product Name field in main.wxs and be shown throughout the setup wizard. This is clearly not the case.

Sorry for the confusion, I feel like I have wasted your time on this one. Keep up the great work and thanks again for cargo-wix.

**Comment by volks73** [comment]
>Although not specified in the documentation, I had expected the name option to override the Product Name field in main.wxs and be shown throughout the setup wizard. This is clearly not the case.

While the overall issue appears to be a "won't fix/no fix needed" situation, maybe you could recommend some documentation or direction to eliminate/reduce confusion?

Do you have a recommendation for an alternative name for the `-n,--name` option that would reduce confusion?

Just to expand and explain a little bit in case someone else comes across this issue: the Product Name is similar to the Manufacturer. I assumed it would not be something that changes too often, or at least not every time an installer (MSI) is created. I view the Product Name similar to the `name` field for the Cargo package, like Manufacturer is related to the `author` field. The `-n,--name` option is for changing the name of the MSI file if the binary or library (crate) name defined in the Cargo.toml is not really a "marketable" product. For example, a binary might be called `name = "server"` and another binary called `name = "client"`. These are pretty generic and a `server-1.0.0-x86_64.msi` and `client-1.0.0-x86_64.msi` is not helpful to a user in identifying the "product" before starting the installer, where Product Name is used. Thus, the `-n,--name` option exists provide the developer flexibility in defining a more descriptive/identifiable name for the eventual MSI file. The product name, something like "Super Awesome Software Suite", is still appropriate within the installer but different from the MSI file name.

**Comment by StevePER** [comment]
> Do you have a recommendation for an alternative name for the -n,--name option that would reduce confusion?
I think using `-o,--output` would be more standard.

> maybe you could recommend some documentation or direction to eliminate/reduce confusion?
The documentation is already really good. I just need to read it better. :) Perhaps you could reiterate at the end of the examples section that once main.wxs has been generated satisfactorily, further customisation will nearly always be required by editing it. I think I was trying too hard to customise as much as possible via command line and Cargo.toml.

**Comment by volks73** [comment]
>I think using -o,--output would be more standard.

I could see adding a `-o,--output` option that would be used to completely override the file name scheme (`<NAME>-<VERSION>-<ARCH>.msi`) with a completely different name and/or destination/location, but maybe not for changing the `<NAME>` component of the default file naming scheme. I will create a new issue to discuss further.

>The documentation is already really good. I just need to read it better. :) Perhaps you could reiterate at the end of the examples >section that once main.wxs has been generated satisfactorily, further customisation will nearly always be required by editing it. I >think I was trying too hard to customise as much as possible via command line and Cargo.toml.

Thanks. I work update the documentation as your have recommended. I will create a new issue to track updating the documentation and close this one as it appears to be at its conclusion. Thank you for the recommendations and information.
